### PR TITLE
systemd: Work around incomplete BLS transition on RHEL 8.0

### PR DESCRIPTION
**Pull request by martinpitt**
grubby on RHEL 8.0 does not change the default kernel args
(https://bugzilla.redhat.com/show_bug.cgi?id=1690765), so that option
changes applied to the current kernels don't apply to future upgrades.

Add the kernel team recommended workaround: If grubenv has kernelopts,
and grubby does not change it by itself, change it manually.

Add a test to TestSystemInfo.testCPUSecurityMitigations that simulates
installing a new kernel, and ensures that this has the expected new
option.

### CODE REVIEW

**Comment by martinpitt** [comment]
Still fails on rhel-7-7/centos-7.

**Comment by martinpitt** [comment]
RHEL/CentOS 7's kernel packages use `new-kernel-pkg` instead of `kernel-install` in their RPM scripts, adjusted the test accordingly. Pushed a fix.

**Comment by martinpitt** [comment]
The cpu security test succeeded everywhere, just respinning for the greener optics. This is ready for review now.

File: pkg/systemd/kernelopt.sh [hunk 1]
```diff
@@ -29,9 +29,23 @@ grub() {
...
         else
             grubby --remove-args="$2" --update-kernel=ALL
+            envopts=$(grub2-editenv - list | grep ^kernelopts) || envopts=""
+            if [ -n "$envopts" ]; then
+                newenvopts=$(echo "$envopts" | sed -r "s/$key(=[^[:space:]\"]*)?//g")
+            fi
         fi
+
+        if [ -n "$envopts" ] && [ "$newenvopts" != "$envopts" ]; then
+            grub2-editenv - set "$newenvopts"
```
Comment (Gundersanne) [inline pkg/systemd/kernelopt.sh]: Just to make sure I understand this correctly, RHEL8 does allow setting of kernel args, but only through the environment?

Comment (martinpitt) [reply pkg/systemd/kernelopt.sh]: It's a little more complicated: In *principle*, RHEL 8 uses BLS, i. e. the initial kernel in a fresh install uses `options $kernelopts` in BLS entries (/boot/loader/entries/*.conf) and `kernelopts` is defined in grubenv. In Fedora 30 (where BLS is fully implemented), `grubby --args=foo --update-kernel=ALL` adds `foo` to grubenv, and leaves all BLS entries alone. But in RHEL 8.0, that command "expands" `options` in the BLS entries with the new option, does *not* update grubenv, and subsequent kernel installs copy that from the running kernel -- i. e. future kernels will miss the option changes. You can try that on our fedora-30 and rhel-8-0 images. This was discussed in the referenced bugzilla.

So, RHEL 8 (as well as Fedora 30 and BLS in general) allows setting options through BLS entries and through grubenv's `$kernelopts`, but the tools don't work well together and produce a mix-and-match.

Comment (Gundersanne) [reply pkg/systemd/kernelopt.sh]: Right, I tihnk i understand it, thanks for the explanation! :)

**Pull request merged**